### PR TITLE
[v6r16] PilotsLogging DB/Service/Client (re-added to integration)

### PR DESCRIPTION
**Pull request by miloszz**
DB, Service and Client for handling extended pilot logging.

Includes start of transition of PilotAgentsDB into SQLAlchemy.

Goes together with [v6r14] Pilot logger first part ( #2319 )

### CODE REVIEW

File: WorkloadManagementSystem/DB/PilotAgentsDB.py [hunk 1]
```diff
@@ -31,15 +31,191 @@
...
+      try:
+        PilotsLogging.__table__.create( self.engine )
+      except Exception, e:
+        return S_ERROR(e)
+    else:
+      gLogger.debug("Table PilotsLogging exists")
+
+##########################################################################################
+  def addPilotsLogging(self, pilotUUID, status, minorStatus, timeStamp, source):
+    "Add new pilot logging entry"
```
Comment (fstagni) [inline WorkloadManagementSystem/DB/PilotAgentsDB.py]: Minor comment: for docstrings always use triple quoters https://www.python.org/dev/peps/pep-0257/ (with """ ... """)

**Comment by fstagni** [comment]
A minor comment: there's an excess of using
```
 except Exception, e
```
which is catching a too generic exception instead of a specific one

File: WorkloadManagementSystem/Client/PilotAgentsClient.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+""" Class that contains client access to the PilotAgents handler. """
+########################################################################
+# $Id$
+# $HeadURL$
+########################################################################
+__RCSID__ = "$Id$"
+
+from DIRAC.Core.Base.Client                         import Client
+
+class PilotAgentsClient( Client ):
```
Comment (fstagni) [inline WorkloadManagementSystem/Client/PilotAgentsClient.py]: This is another minor comment: I believe a better name would be "PilotsLoggingClient", as "PilotAgents" is an obsolete naming

**Comment by chaen** [comment]
And btw, it should be "except Exception as e" now ;-)

File: WorkloadManagementSystem/Service/PilotAgentsHandler.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  def initialize(self):
+    self.pilotAgents = PilotAgentsDB()
+    
+  auth_addPilotsLogging = [ 'all' ]
+  types_addPilotsLogging = [ types.StringType, types.StringType, types.StringType, types.FloatType, types.StringType ]
+  def export_addPilotsLogging( self, pilotUUID, status, minorStatus, timeStamp, source ):
+    
+    return self.pilotAgents.addPilotsLogging( pilotUUID, status, minorStatus, timeStamp, source )
+  
+  auth_getPilotsLogging = [ 'all' ]
```
Comment (fstagni) [inline WorkloadManagementSystem/Service/PilotAgentsHandler.py]: The auth rules should be revised here. 

I would say that "authenticated" is the minimal authorization level, but for "set" and especially "delete" methods we should have "Operator" as base.

**Comment by fstagni** [comment]
There's a missing change, the one for WorkloadManagement/ConfigTemplate.cfg file, in which the PilotAgents service (which again I would call "PilotsLoggingHandler", so "PilotsLogging" service, is added.

**Pull request closed without merging**